pkg/provider/helm/helm.go

```diff
@@ -45,35 +45,69 @@ var _ Provider = &HelmSPIREProvider{}
 // helm-charts-hardened Helm chart to install a SPIRE stack to a given Kubernetes context, making use of the Cofide
 // API concepts and abstractions
 type HelmSPIREProvider struct {
-	ctx              context.Context
-	settings         *cli.EnvSettings
-	cfg              *action.Configuration
-	SPIREVersion     string
-	SPIRECRDsVersion string
-	spireValues      map[string]any
-	spireCRDsValues  map[string]any
-	trustZoneName    string
-	cluster          *clusterpb.Cluster
+	ctx                 context.Context
+	settings            *cli.EnvSettings
+	cfg                 *action.Configuration
+	SPIREVersion        string
+	SPIRECRDsVersion    string
+	spireValues         map[string]any
+	spireCRDsValues     map[string]any
+	trustZoneName       string
+	cluster             *clusterpb.Cluster
+	SPIRERepositoryURL  string
+	SPIRERepositoryName string
 }
 
-func NewHelmSPIREProvider(ctx context.Context, trustZoneName string, cluster *clusterpb.Cluster, spireValues, spireCRDsValues map[string]any, kubeConfig string) (*HelmSPIREProvider, error) {
+// HelmSPIREProviderOption is a function that configures a HelmSPIREProvider.
+type HelmSPIREProviderOption func(*HelmSPIREProvider)
+
+// WithKubeConfig sets the kubeconfig path
+func WithKubeConfig(kubeConfig string) HelmSPIREProviderOption {
+	return func(p *HelmSPIREProvider) {
+		if kubeConfig != "" {
+			p.settings.KubeConfig = kubeConfig
+		}
+	}
+}
+
+// WithSPIRERepositoryURL sets the SPIRE Helm repository URL
+func WithSPIRERepositoryURL(url string) HelmSPIREProviderOption {
+	return func(p *HelmSPIREProvider) {
+		if url != "" {
+			p.SPIRERepositoryURL = url
+		}
+	}
+}
+
+// WithSPIRERepositoryName sets the name for the SPIRE Helm repository
+func WithSPIRERepositoryName(name string) HelmSPIREProviderOption {
+	return func(p *HelmSPIREProvider) {
+		if name != "" {
+			p.SPIRERepositoryName = name
+		}
+	}
+}
+
+func NewHelmSPIREProvider(ctx context.Context, trustZoneName string, cluster *clusterpb.Cluster, spireValues, spireCRDsValues map[string]any, opts ...HelmSPIREProviderOption) (*HelmSPIREProvider, error) {
 	settings := cli.New()
 	settings.KubeContext = cluster.GetKubernetesContext()
 	settings.SetNamespace(SPIREManagementNamespace)
 
-	if kubeConfig != "" {
-		settings.KubeConfig = kubeConfig
+	prov := &HelmSPIREProvider{
+		ctx:                 ctx,
+		settings:            settings,
+		SPIREVersion:        SPIREChartVersion,
+		SPIRECRDsVersion:    SPIRECRDsChartVersion,
+		spireValues:         spireValues,
+		spireCRDsValues:     spireCRDsValues,
+		trustZoneName:       trustZoneName,
+		cluster:             cluster,
+		SPIRERepositoryURL:  SPIRERepositoryURL,
+		SPIRERepositoryName: SPIRERepositoryName,
```

Comment:
We should use the new Options here, and pass these (as with the KubeConfig) via the factory in this case

Reply:
We use a custom `SPIRERepositoryURL` and `SPIRERepositoryName` in the Connect plugin, using the provider factory.